lib/animations/animated_switcher/animated_switcher.dart

```diff
@@ -0,0 +1,238 @@
+import 'dart:math';
+
+import 'package:flutter/material.dart';
+
+import '../../contributors_card.dart';
+
+class AnimatedSwitcherScreen extends StatefulWidget {
+  const AnimatedSwitcherScreen({Key? key}) : super(key: key);
+
+  @override
+  State<AnimatedSwitcherScreen> createState() => _AnimatedSwitcherScreenState();
+}
+
+class _AnimatedSwitcherScreenState extends State<AnimatedSwitcherScreen> {
+  late bool _isFavorite;
+  late bool _isRotated;
+  late int _currentIndex;
+
+  late List<Widget> widgets;
+
+  @override
+  void initState() {
+    super.initState();
+    _isFavorite = false;
+    _isRotated = false;
+    _currentIndex = 0;
```

Comment:
Optional: It's better to avoid using the `late` keyword unless absolutely necessary (this is done to avoid unexpected errors if we forget to provide the late variable a value). Although it won't make a huge difference in this case, it would just set a good practice for all engineers.

Note: This same comment is applicable for the `widgets` variable as well. It's not being used as a state (i.e. it's not updated anywhere in the code), therefore, it should be alright to inline the variable initialisation.